sources/web/datalab/polymer/components/item-list/item-list.ts

```diff
@@ -62,39 +65,103 @@ class ItemListElement extends Polymer.Element {
         value: function(): Array<Object> {
           return [];
         },
+        observer: '_rowsChanged',
       },
       columns: {
         type: Array,
         value: function(): Array<string> {
           return [];
         }
+      },
+      _selectedElements: {
+        type: Array,
+        value: function(): Array<Object> {
+          return [];
+        }
       }
     }
   }
 
   /**
-   * on row click, check the click target, if it's the checkbox, add it to
-   * the selected rows, otherwise select it only
+   * Returns list of currently selected elements. This list keeps the actual
+   * HTML elements, which can then be used to get their indices, whereas the
+   * opposite is not directly possible.
+   */
+  getSelectedElements() {
+    return this._selectedElements;
+  }
+
+  /**
+   * Returns list of indices for the currently selected elements.
+   */
+  getSelectedIndices() {
+    return this._selectedElements.map(element => {
+      return this.$.list.indexForElement(element);
+    });
+  }
+
+  /**
+   * When the list of rows is refreshed, no items should be selected.
```

Comment:
Perhaps you can figure out a way to reword this comment so it says what this function does ("clears the list of selected items" or some such) as opposed to the desired state after it runs.